source/fundamentals/indexes.txt

```diff
@@ -0,0 +1,253 @@
+:manual:`Data Modeling and Indexes </core/data-model-operations/#data-model-indexes>`.
+
+Index Types
+-----------
+
+MongoDB supports a number of different index types to support querying your data. The following sections describe the
+most common index types and provide sample code for creating each index type.
+
+.. tip::
+
+   The MongoDB Java Driver provides the :java-core-api:`Indexes </com/mongodb/client/model/Indexes.html>` class that
+   includes static factory methods to create index specification documents for different MongoDB Index key types.
+
+Single Field and Compound Indexes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Single Field Indexes
+++++++++++++++++++++
+
+**Single field indexes** are indexes that improve performance for queries that specify ascending or descending sort
+order on a single field of a document.
+
+The following example uses the ``createIndex()`` method to create an ascending order index on the ``title`` field in the
+``movies`` collection in the ``sample_mflix`` database.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/IndexPage.java
+   :language: java
+   :dedent:
+   :start-after: begin single index
+   :end-before: end single index
+
+The following is an example of a query that would be covered by the index created above.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/IndexPage.java
+   :language: java
+   :dedent:
+   :start-after: begin covered single query
+   :end-before: end covered single query
+
+See the MongoDB server manual section on :manual:`single field indexes </core/index-single>` for more information.
+
+Compound Indexes
+++++++++++++++++
+
+**Compound indexes** are indexes that improve performance for queries that specify ascending or descending sort order
+for *multiple* fields of a document. You must specify the direction (ascending or descending) for each field in the
+index.
+
+The following example uses the ``createIndex()`` method to create a compound index on the ``type`` and ``rated`` fields
+in the ``movies`` collection in the ``sample_mflix`` database.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/IndexPage.java
+   :language: java
+   :dedent:
+   :start-after: begin compound index
+   :end-before: end compound index
+
+The following is an example of a query that would be covered by the index created above.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/IndexPage.java
+   :language: java
+   :dedent:
+   :start-after: begin covered compound query
+   :end-before: end covered compound query
+
+
+See the MongoDB server manual section on :manual:`Compound indexes </core/index-compound>` for more information.
+
+Multikey Indexes (Indexes on Array Fields)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Multikey indexes** are indexes that improve performance for queries that specify ascending or descending indexes on
+fields that contain an array value. You can define a multikey index using the same syntax as a single field or compound
```

Comment:
Personal opinion here: I think this sentence gets a bit lost in the weeds of "queries that specify ascending or descending indexes" to a point where it's easy to miss the actual idea (multikey indexes are indexes on fields that contain an array value). Arguably you've mitigated this issue in the title of this subsection, but I wonder if we could trim this sentence a little to make it less wordy.